pubsub/tests/test_asserts.go

```diff
@@ -92,3 +93,19 @@ func AssertMessagesMetadata(t *testing.T, key string, expectedValues map[string]
 
 	return ok
 }
+
+// AssertAllMessagesHaveSameContext checks if context of all received messages is the same as in expectedValues, if PreserveContext is enabled.
+func AssertAllMessagesHaveSameContext(t *testing.T, contextKeyString string, expectedValues map[string]context.Context, received []*message.Message) bool {
+	assert.Len(t, received, len(expectedValues))
+
+	ok := true
+	for _, msg := range received {
+		expectedValue := expectedValues[msg.UUID].Value(contextKey(contextKeyString)).(string)
+		actualValue := msg.Context().Value(contextKeyString)
+		if !assert.Equal(t, expectedValue, actualValue) {
+			ok = false
+		}
+	}
+
+	return ok
+}
```

Comment:
You don't need to return a boolean, calling t.Error is enough to invalidate the test

also your code is not using the returned value when you call it

So this would be better I think

```suggestion
func AssertAllMessagesHaveSameContext(t *testing.T, contextKeyString string, expectedValues map[string]context.Context, received []*message.Message) {
	assert.Len(t, received, len(expectedValues))

	for _, msg := range received {
		expectedValue := expectedValues[msg.UUID].Value(contextKey(contextKeyString)).(string)
		actualValue := msg.Context().Value(contextKeyString)
		assert.Equal(t, expectedValue, actualValue)
	}
}
```

Reply:
Agreed, changed
Thanks for the suggestion